float32 x
float32 y
float32 v
float32 theta
float32 r
float32 deltaTime